server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSOptionsCheckBaseUtility.java

```diff
@@ -277,14 +277,16 @@ private void getAllTokenChildren(
     if (ctx.children == null) return;
     ctx.children.forEach(
         child -> {
-          if (TerminalNode.class.isAssignableFrom(child.getClass())
-              && baseDuplicateOptions.containsKey(child.getText().toUpperCase()))
-            children.add((TerminalNode) child);
-          else if (ParserRuleContext.class.isAssignableFrom(child.getClass())) {
-            if (validateResponseHandler
-                && child.getClass().getSimpleName().equals("Cics_handle_responseContext"))
-              checkResponseHandlers((CICSParser.Cics_handle_responseContext) child);
-            getAllTokenChildren((ParserRuleContext) child, children, validateResponseHandler);
+          if (!child.getText().startsWith("(")) {
```

Comment:
Without that this would fail
```
           EXEC CICS READ
                DATASET   (WS-ACCTDAT-FILE)
                INTO      (ACCOUNT-RECORD)
                LENGTH    (LENGTH OF ACCOUNT-RECORD)
                RIDFLD    (ACCT-ID)
                KEYLENGTH (LENGTH OF ACCT-ID)
                UPDATE
                RESP      (WS-RESP-CD)
                RESP2     (WS-REAS-CD)
           END-EXEC
```

its part of server/engine/src/test/java/org/eclipse/lsp/cobol/positive/PositiveTest.java

Reply:
I think that the checking (and possibly rules) will have to be re-organized a bit, so that we don't have to recursivly check every token.
Apparently, we may also encounter `DATASET(DATASET)` situation.

As a temporary fix, I think we could just avoid recursing into `cics_data_area`, `cics_name`, and maybe `cics_data_value` rules.